recipes/libpqxx/all/conanfile.py

```diff
@@ -25,102 +28,102 @@ class LibpqxxConan(ConanFile):
-        return "build_subfolder"
+    def _compilers_minimum_version(self):
+        if Version(self.version) < "7.0":
+            return {
+                "gcc": "7",
+                "clang": "6",
+                "apple-clang": "10"
+            }
+        else:
+            return {
+                "gcc": "7" if Version(self.version) < "7.5.0" else "8",
+                "clang": "6",
+                "apple-clang": "10"
+            }
 
     @property
     def _mac_os_minimum_required_version(self):
         return "10.15"
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("libpq/14.2")
+        self.requires("libpq/14.5")
 
     def validate(self):
-        if self.options.shared and msvc_runtime_flag(self) == "MTd":
-            raise ConanInvalidConfiguration(
-                "{} recipes does not support build shared library with MTd runtime."
-                .format(self.name,))
-
-        compiler = str(self.settings.compiler)
-        compiler_version = tools.Version(self.settings.compiler.version)
-
-        lib_version = tools.Version(self.version)
-        lib_version_7_6_0_or_later = lib_version >= "7.6.0"
-        minimum_compiler_version = {
-            "Visual Studio": "16" if lib_version_7_6_0_or_later else "15",
-            "msvc": "192" if lib_version_7_6_0_or_later else "191",
-            "gcc": "8" if lib_version >= "7.5.0" else "7",
-            "clang": "6",
-            "apple-clang": "10"
-        }
-
-        minimum_cpp_standard = 17
-
-        if compiler in minimum_compiler_version and \
-           compiler_version < minimum_compiler_version[compiler]:
-            raise ConanInvalidConfiguration("{} requires a compiler that supports"
-                                            " at least C++{}. {} {} is not"
-                                            " supported."
-                                            .format(self.name, minimum_cpp_standard, compiler, compiler_version))
-
-        if self.settings.os == "Macos":
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        if Version(self.version) < "7.0":
+            check_min_vs(self, 190)
+        elif Version(self.version) < "7.6":
+            check_min_vs(self, 191)
+        else:
+            check_min_vs(self, 192)
+
+        if is_msvc(self) and self.options.shared and self.settings.compiler.runtime == "MTd":
```

Comment:
`compiler.runtime` is specific to `Visual Studio`. Can you use [conan.tools.microsoft.msvc_runtime_flag()](https://docs.conan.io/en/latest/reference/conanfile/tools/microsoft.html#conan-tools-microsoft-msvc-runtime-flag) which supports both `Visual Studio` and `msvc` to check against the `MTd` flag here?